conftest.py

```diff
@@ -43,7 +43,14 @@ def get_test_dtypes() -> Dict[str, torch.dtype]:
 
 @pytest.fixture()
 def device(device_name) -> torch.device:
-    if device_name not in TEST_DEVICES:
+    # To work with TPUs on Colab:
+    # pip install cloud-tpu-client==0.10
+    # pip install https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.6-cp36-cp36m-linux_x86_64.whl
+    # Issue: gradcheck on TPUs takes ages...
+    if device_name == 'tpu':
+        import torch_xla.core.xla_model as xm
+        return xm.xla_device()
+    elif device_name not in TEST_DEVICES:
         pytest.skip(f"Unsupported device type: {device_name}")
```

Comment:
Should this return None?

Reply:
Try something like:
```bash
$ pytest -vv -r sxf test/ --device tpu --dtype float64
```
to show the skipped reason. I would suspect it falls into ```pytest.skip(f"Unsupported device type: {device_name}")```